pkg/collector/scheduler/scheduler.go

```diff
@@ -126,6 +130,9 @@ func (s *Scheduler) Cancel(id check.ID) error {
 	delete(s.checkToQueue, id)
 
 	schedulerChecksEntered.Add(-1)
+	if checkName, ok := s.tlmTrackedChecks[id]; ok {
```

Comment:
Should the check also be removed from `tlmTrackedChecks`?

Reply:
you're right, I'll address

Reply:
addressed, ready for another review @truthbk